<a href="https://colab.research.google.com/github/gustaavoq94/proyecto2/blob/main/proyecto2.2" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [4]:
filename = '/content/day.csv'
df = pd.read_csv(filename)
df.head()

instant      dteday  season  yr  mnth  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1        0        6           0   
1        2  2011-01-02       1   0     1        0        0           0   
2        3  2011-01-03       1   0     1        0        1           1   
3        4  2011-01-04       1   0     1        0        2           1   
4        5  2011-01-05       1   0     1        0        3           1   

   weathersit      temp     atemp       hum  windspeed  casual  registered  \
0           2  0.344167  0.363625  0.805833   0.160446     331         654   
1           2  0.363478  0.353739  0.696087   0.248539     131         670   
2           1  0.196364  0.189405  0.437273   0.248309     120        1229   
3           1  0.200000  0.212122  0.590435   0.160296     108        1454   
4           1  0.226957  0.229270  0.436957   0.186900      82        1518   

    cnt  
0   985  
1   801  
2  1349  
3  1562  
4  1600

#**Fuente de datos**
https://www.kaggle.com/datasets/ayessa/bike-sharing-dataset-regression

#**Breve descripción de datos**
El conjunto de datos "Bike Rent" contiene información sobre el alquiler de bicicletas, incluyendo detalles sobre la fecha y hora del alquiler, el clima, la temperatura y la cantidad de alquileres realizados. Cada fila representa un registro de alquiler, y las columnas incluyen información como el momento en que se registró el alquiler (instant), la fecha y hora (dteday), el período de tiempo (season), el año (yr), el mes (mnth), si el día es un feriado (holiday), el día de la semana (weekday), si es un día laborable o no (workingday), la condición meteorológica (weathersit), la temperatura (temp), la sensación térmica (atemp), la humedad (hum), la velocidad del viento (windspeed) y la cantidad total de alquileres (cnt), que se divide en la cantidad de alquileres de usuarios registrados (registered) y la cantidad de alquileres de usuarios casuales (casual).

#**Cual es el objetivo?**
"cnt" sería una buena opción para usar como objetivo ya que representa la cantidad total de alquileres de bicicletas realizados en un día, y es una variable numérica continua que puede ser utilizada para construir modelos de regresión.

#**¿Este es un problema de clasificación o regresión?**
Regresion

In [5]:
df.shape

(731, 16)

#**¿Cuántas características hay?**
16
#**¿Cuántas filas de datos hay?**
731

#**¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?**
El conjunto de datos "Bike Rent" es relativamente limpio y bien estructurado, lo que hace que la tarea de limpieza y exploración sea relativamente sencilla. Sin embargo, aún existen algunos retos que podrían presentarse durante estas etapas

*Valores faltantes*: aunque el conjunto de datos no parece tener muchos valores faltantes, es posible que algunas columnas tengan valores faltantes que deban ser tratados. Dependiendo del número de valores faltantes y de la columna afectada, es posible que se deba tomar una decisión sobre cómo manejar estos valores faltantes (eliminar las filas con valores faltantes, imputar valores, etc.).

*Datos atípicos*: es posible que el conjunto de datos tenga valores atípicos o extremos que deban ser identificados y tratados. Estos valores pueden tener un impacto significativo en los resultados del modelo, por lo que es importante prestar atención a ellos durante la exploración y limpieza de los datos.

*Correlación entre variables*: algunas de las variables del conjunto de datos pueden estar altamente correlacionadas, lo que puede tener un impacto en el rendimiento del modelo. Por ejemplo, la variable "temp" y la variable "atemp" pueden estar altamente correlacionadas ya que ambas miden la temperatura, pero de diferentes maneras. Es importante identificar estas correlaciones y tomar medidas para manejarlas, si es necesario.

*Selección de características*: cuando se construye un modelo de machine learning, es importante seleccionar las características adecuadas para el modelo. En el caso del conjunto de datos "Bike Rent", es posible que algunas características no sean relevantes para el modelo y deban ser eliminadas. Por ejemplo, la columna "instant" no parece ser relevante para el modelo ya que simplemente identifica el registro de alquiler y no tiene ninguna relación con la cantidad de alquileres de bicicletas.